src/python/vdo-setup-dmtest/vdo-setup-dmtest.py

```diff
@@ -0,0 +1,512 @@
+      osVersion = result.stdout.rstrip()
+    elif (os.path.exists('/etc/centos-release')):
+      result = runCommand('cut -d " " -f4 /etc/centos-release')
+      osVersion = result.stdout.rstrip()
+
+    if (int(osVersion) > 9):
+      osVersion = 9
+
+    branch = 'rhel-' + str(osVersion)
+    assert branch in remoteBranches, " ".join(['Invalid remote branch', branch])
+
+    # Build and install the module for the running kernel
+    repo.git.switch(branch)
+
+    # Check for a module signing certificate/key pair, or create one
+    certsDir = os.path.join('/lib/modules', os.uname().release, 'build/certs')
+    certPath = os.path.join(certsDir, 'signing_key.pem')
+
+    self.log.info('Checking for signing certificate {0}'.format(certPath))
+    certExists = os.path.exists(certPath)
+    if (certExists):
+      certValid = verifyCertificate(certPath)
+
+    if not certExists or not certValid:
+      # Check for a x509.genkey file in the certs directory, or create one
+      genkeyPath = os.path.join(certsDir, 'x509.genkey')
+      if not (os.path.exists(genkeyPath)):
+        assert createX509Genkey(certsDir) == True, 'Unable to create x509.genkey'
+
+      self.log.info("Creating the certificate/key pair needed for signing the module")
+      keyPath = os.path.join(certsDir, 'signing_key.x509')
+      opensslCmd = ['sudo openssl req -new -nodes -utf8 -sha512 -days 36500 -batch -x509',
+                    '-config', genkeyPath, '-outform PEM -out', certPath, '-keyout', keyPath]
+      result = runCommand(opensslCmd)
+      assert result.returncode == 0, "Creating OpenSSL certificate failed"
+
+    commands = ['make -C /lib/modules/$(uname -r)/build M=$PWD',
+                'sudo make -C /lib/modules/$(uname -r)/build M=$PWD modules_install',
+                'sudo modprobe ' + moduleName]
+    for cmd in commands:
+      result = runCommand(cmd)
+      assert result.returncode == 0, 'Non-zero return status'
+
+  def installBlkArchive(self, repoName):
+    """
+    Perform installation steps for the blk-archive repo
+    """
+    # Skip if already installed
+    userCargoPath = os.path.join(os.path.expanduser('~'), '.cargo/bin')
+    localExePath = os.path.join(os.getcwd(), 'target/release', repoName)
+    cargoExePath = os.path.join(userCargoPath, repoName)
+    if isExecutable(localExePath) and isExecutable(cargoExePath):
+      self.log.debug("{0} is already installed - skipping".format(repoName))
+      if userCargoPath not in os.environ["PATH"]:
+        self.log.warning("PATH must include '{path}' for {repo} to function properly".format(
+          path = userCargoPath, repo = repoName))
+
+        # Append strings to necessary lists for reporting downstream
+        self.envVars.append('PATH=$PATH' + os.pathsep + userCargoPath)
+        self.dmtestHealthWarn.append('Dependency blk-archive will display as missing until PATH' \
+                                     + ' is updated')
```

Comment:
Perhaps these strings can be created without + concatenation?

```
self.envVars.append('PATH=$PATH{pathsep}{userCargoPath}'.format(pathsep=os.pathsep, userCargoPath=userCargoPath))
self.dmtestHealthWarn.append('Dependency blk-archive shows as missing until PATH is updated')
```

Reply:
Fixed in latest push.